lib/ts/recipe/totp/recipeImplementation.ts

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2023, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RecipeInterface, TypeNormalisedInput } from "./types";
+import { Querier } from "../../querier";
+import NormalisedURLPath from "../../normalisedURLPath";
+
+export default function getRecipeInterface(querier: Querier, config: TypeNormalisedInput): RecipeInterface {
+    function copyAndRemoveUserContext(input: any): any {
+        let result = {
+            ...input,
+        };
+        delete result.userContext;
+        return result;
+    }
+
+    return {
+        createDevice: async function (input) {
+            const { userContext, userIdentifierInfo, ...rest } = input;
```

Comment:
need to add logic for coming up with the name in case the user has not given it

Reply:
Not here. In the API implementation, right?